dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/TrnTokenSignInJourney.cs

```diff
@@ -60,8 +67,14 @@ public override bool CanAccessStep(string step)
         (CoreSignInJourney.Steps.Phone, _) => Steps.Landing,
         (CoreSignInJourney.Steps.PhoneConfirmation, _) => CoreSignInJourney.Steps.Phone,
         (CoreSignInJourney.Steps.ResendPhoneConfirmation, _) => CoreSignInJourney.Steps.PhoneConfirmation,
-        (Steps.CheckAnswers, { MobileNumberVerified: true }) => CoreSignInJourney.Steps.Phone,
+        (CoreSignInJourney.Steps.Email, { ContactDetailsVerified: true }) => Steps.CheckAnswers,
+        (CoreSignInJourney.Steps.Email, { MobileNumberVerified: false }) => CoreSignInJourney.Steps.PhoneConfirmation,
+        (CoreSignInJourney.Steps.Email, { MobileNumberVerified: true }) => CoreSignInJourney.Steps.Email,
```

Comment:
This is slightly awkward but avoids the situation where going back to check-answers or phone-confirmation is not valid. There may be a better way to implement the back link by doing something similar to getLastAccessibleStep